maven-release-manager/src/main/java/org/apache/maven/shared/release/phase/GenerateReleasePomsPhase.java

```diff
@@ -543,9 +543,10 @@ private List<Plugin> createReleasePlugins(ReleaseDescriptor releaseDescriptor, M
                     }
                     releasePlugin.setExecutions(plugin.getExecutions());
                     releasePlugin.setDependencies(plugin.getDependencies());
-                    releasePlugin.setGoals(plugin.getGoals());
                     releasePlugin.setInherited(plugin.getInherited());
-                    releasePlugin.setConfiguration(plugin.getConfiguration());
+                    if (plugin.getConfiguration() != null) {
+                        releasePlugin.setConfiguration(plugin.getConfiguration());
```

Comment:
In 4.x configuration can not be a null

Reply:
why is the if clause required? Does `releasePlugin` have a default, non-null config?

Reply:
Alright, I see!